flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/LocalStandaloneFlinkResourceFactory.java

```diff
@@ -80,6 +80,33 @@ public FlinkResource create(FlinkResourceSetup setup) {
 		return new LocalStandaloneFlinkResource(distributionDirectory.get(), logBackupDirectory.orElse(null), setup);
 	}
 
+	/**
+	 * Utils to find the flink project root directory.
+	 * @param currentDirectory
+	 * @return The flink project root directory.
+	 */
+	public static Path getProjectRootDirectory(Path currentDirectory) {
```

Comment:
We can use this method to replace the code block in `LocalStandaloneFlinkResourceFactory#create()`.